### PR TITLE
expression: migrate test-infra to testify for integration_test.go

### LINKED ISSUE #29139: migrate test-infra to testify for `expression/integration_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #29139

Problem Summary:

### What is changed and how it works?

expression: migrate test-infra to testify for integration_test.go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/integration_test.go [hunk 1]
```diff
@@ -147,22 +103,34 @@ func (s *testIntegrationSuite) Test19654(c *C) {
...
+	require.Contains(t, str, "decimal(16,2)")
 }
 
-func (s *testIntegrationSuite) TestFuncREPEAT(c *C) {
-	tk := testkit.NewTestKit(c, s.store)
-	defer s.cleanEnv(c)
+func TestFuncREPEAT(t *testing.T) {
+	t.Parallel()
+
+	t.Parallel()
```
Comment (tisonkun) [inline expression/integration_test.go:127]:
```suggestion
	t.Parallel()
```

ditto others

Comment (hawkingrei) [reply expression/integration_test.go]: fix the other three. Thanks.

**Comment by hawkingrei** [comment]
@tisonkun @lcwangchao PTAL

**Review by tisonkun — APPROVED** [review]
LGTM

File: expression/integration_test.go [hunk 2]
```diff
@@ -6792,8 +4681,14 @@ func (s *testIntegrationSuite) TestCTEWithDML(c *C) {
...
-func (s *testIntegrationSuite) TestIssue16419(c *C) {
-	tk := testkit.NewTestKitWithInit(c, s.store)
+func TestIssue16419(t *testing.T) {
+	t.Parallel()
+
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
```
Comment (AilinKid) [inline expression/integration_test.go:4691]: duplicate

Comment (AilinKid) [reply expression/integration_test.go:4691]: ditto for the above

Comment (hawkingrei) [reply expression/integration_test.go]: removed the other three.

File: expression/integration_serial_test.go [hunk 3]
```diff
@@ -1491,3 +1502,3096 @@ func TestCollationForBinaryLiteral(t *testing.T) {
 	tk.MustQuery("select * from t where col1 not in (0x1B,0x20) order by col1").Check(testkit.Rows("\x1e \xec 6966939640596047133"))
 	tk.MustExec("drop table t")
 }
+
+func TestMathBuiltin(t *testing.T) {
+	t.Skip("it has been broken. Please fix it as soon as possible.")
```
Comment (AilinKid) [inline expression/integration_serial_test.go:1507]: some skipped test comment can be removed if it's right

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: expression/integration_serial_test.go (hunk 3 above)
Comment (hawkingrei) [reply expression/integration_serial_test.go:1507]: if removed, the CI will fail.

**Review by AilinKid — APPROVED** [review]

**Pull request merged**